repair-tool.lic

```diff
@@ -0,0 +1,108 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#enchant
+=end
+
+custom_require.call(%w[common common-items events common-crafting equipmanager])
+
+class RepairTool
+  include DRC
+  include DRCC
+  include DRCA
+  include DRCI
+
+  def initialize
+    @settings = get_settings
+    @hometown = @settings.hometown
+    @bag = @settings.crafting_container
+
+    # not used, but passed to stow_crafting_item in crafting-common.lic
+    @belt = @settings.crafting_container
+    # not used, but passed to stow_crafting_item in crafting-common.lic
+    @bag_items = @settings.crafting_items_in_container
+
+    crafting_data = get_data('crafting')
+    @recipe_parts = crafting_data['recipe_parts']
+    @discipline = 'blacksmithing'
+    info = crafting_data[@discipline][@hometown]
+    unless info
+      echo("No crafting settings found for discipline: #{discipline}")
+      exit
+    end
+
+    @brush = @settings.repair_tools.find { |item| /brush/ =~ item } || 'brush'
+    @oil = @settings.repair_tools.find { |item| /oil/ =~ item } || 'oil'
+
+    @brush_part_name = "wire brush"
+    @oil_part_name = "flask of oil"
+
+    arg_definitions = [
+      [
+        { name: 'tool', display: 'tool name', regex: /^[A-z\s\-']+$/i, variable: true, description: 'Name of the tool, wrap in double quotes if this is multiple words.' },
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    @tool = args.tool
+
+    @equipment_manager = EquipmentManager.new
+    @equipment_manager.empty_hands
+
+    repair_item
+  end
+
+  def replenish_repair_tools(part)
+    data = @recipe_parts[part][@hometown]
+    order_item(data['part-room'], data['part-number'])
+  end
+
+  # TODO: try using this in repair_item
+  def stow_tool(name)
+    stow_crafting_item(name, @bag, @belt)
+  end
+
+  # TODO: try using this in repair_item
+  def get_tool(name)
+    get_crafting_item(name, @bag, @bag_items, @belt, true)
+  end
+
+  def repair_item
+    waitrt?
+    pause
+    bput "get my #{@tool} from my #{@bag}", "You get", "What were you referring to?"
+    pause
```

Comment:
Why were the `pause`s necessary?

Reply:
+1

Reply:
probably not with bput, they were in there when it was an fput, which did not work, will remove